lib/ovirt/service.rb

```diff
@@ -145,8 +155,48 @@ def status(link)
       node.xpath('status/state').text
     end
 
+    # Checks if the API is available in the given candidate path. It does so sending a request without
+    # authentication. If the API is available there it will respond with the "WWW-Autenticate" header
+    # and with the "RESTAPI" or "ENGINE" realm.
+    def probe_api_path(uri, path)
+      uri = URI.join(uri, path)
+      request = RestClient::Resource.new(uri.to_s, :verify_ssl => OpenSSL::SSL::VERIFY_NONE)
+      begin
+        request.get
```

Comment:
Yes, if the GET, without authentication, is successful that means that we aren't talking to the oVirt REST API, as it always requires authentication.